protoutil/conversion.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package protoutil provides mechanisms for interacting with proto.
+package protoutil
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+// ToProtoStruct converts v, which must marshal into a JSON object, into a proto struct.
+func ToProtoStruct(v any) (*structpb.Struct, error) {
```

Comment:
@yolocs should we genericify this? 

```go
func ToProto[P proto.Message](v any) (P, error)
```

That might not actually work, but it would give us a generic function to convert any thing to a specific proto type.

Reply:
```go
func ToProto[T proto.Message](v any) (T, error) {
	var t T

	jb, err := json.Marshal(v)
	if err != nil {
		return t, fmt.Errorf("json.Marshal: %w", err)
	}
	if err := json.Unmarshal(jb, &t); err != nil {
		return t, fmt.Errorf("failed to unmarshal json: %w", err)
	}
	return t, nil
}
```

Reply:
func ToProto[T proto.Message](v any) (T, error) {
	var t T

	jb, err := json.Marshal(v)
	if err != nil {
		return t, fmt.Errorf("json.Marshal: %w", err)
	}
	if err := json.Unmarshal(jb, &t); err != nil {
		return t, fmt.Errorf("failed to unmarshal json: %w", err)
	}
	return t, nil
}

Reply:
First I don't think this code would compile :) because proto.Message is an interface (remember our earlier discussion?)

I don't see much value to make this generic. Proto struct works because you can unmarshal pretty much any json object into it. But that's not true for other proto types.